### DISCUSSION TITLE
unit testing in polyglot (node) packages

### DISCUSSION THREAD

**Discussion by pe1pip** [vmware/build-tools-for-vmware-aria#438] [Q&A]
Hi,

I'm looking into node.js polyglot code for orchestrator and while there is a lot missing from these runtimes (no config elements or resource elements, or any other form of persistent storage) there might be benefits. But, for normal javascript and  typescript we have jasmine to perform unit tests. Has anyone done something similar (maybe even JEST?) for polyglot (node.js) packages?

**Comment by Michaelpalacce** [comment]
Haven't heard of anything like this before. @VenelinBakalov have you encountered this?

**Comment by VenelinBakalov** [comment]
No but let me forward the question to the team and see what comes up.

**Reply by pe1pip** [reply]
Hi Guys,

I've been sort of playing with a few options and the good news is that it's not very hard to get bare-bones JEST working. There are a few downsides:

- the dev dependencies have to be cleaned up from the package node_modules
- other dependencies do have to be installed
- .test.ts files and .helper.ts files are not ignored by the packager so they cause all sorts of errors during transpilation

I think there are ways we can work around all of that, I'll tinker a bit more if I have time.

**Reply by VenelinBakalov** [reply]
@pboychev-bcom just FYI since I know you were as well researching into jest topic

**Comment by VenelinBakalov** [comment]
Currently there is no support for unit testing in polyglot packages.
